src/syscall_hook.h

```diff
@@ -0,0 +1,57 @@
+#ifndef CORE_SYSCALL_HOOK_H
+#define CORE_SYSCALL_HOOK_H
+
+#define _GNU_SOURCE
+#include <dlfcn.h>
+#include <errno.h>
+#include <sys/socket.h>
+#include <sys/uio.h>
+#include <unistd.h>
+
+typedef int (*sys_connect)(int sockfd,
+                           const struct sockaddr *addr,
+                           socklen_t addrlen);
+typedef int (*sys_accept)(int sockfd,
+                          struct sockaddr *addr,
+                          socklen_t *addrlen);
+
+typedef ssize_t (*sys_read)(int fd, void *buf, size_t count);
+typedef ssize_t (*sys_recv)(int sockfd, void *buf, size_t len, int flags);
+
+typedef ssize_t (*sys_write)(int fd, const void *buf, size_t count);
+typedef ssize_t (*sys_send)(int sockfd, const void *buf, size_t len, int flags);
+
+#define HOOK_SYSCALL(name) \
+    real_sys_##name = (sys_##name) dlsym(RTLD_NEXT, #name)
+
+#define CONN_TIMEOUT (10 * 1000)
```

Comment:
The macro `CONN_TIMEOUT`, `ACCEPT_TIMEOUT`, `READ_TIMEOUT`, and `RECV_TIMEOUT` are supposed to be implementation specific. However, your modifications violate the encapsulation.